### ISSUE TITLE
Slidedeck for the MVP engineering onboarding meeting

### ISSUE THREAD

**Issue by hpjaj** [department-of-veterans-affairs/va.gov-team#287]
## Background

As part of the [epic for preparing to onboard our first VFS Team](https://github.com/department-of-veterans-affairs/va.gov-team/issues/202) we will conducting a breakout meeting for engineering onboarding.

Here is the PR containing the engineering best practices "talk points" that will be included in the presentation:

https://github.com/department-of-veterans-affairs/va.gov-vfs-teams/pull/15

## Definition of done

- [x] slidedeck for engineering onboarding meeting
- [x] an agenda that can be distributed in advance of the meeting
- [x] links to resources discussed in the onboarding meeting for sharing with VFS team members

**Comment by hpjaj** [comment]
@gunsch @rroueche @rtluu @accbjt - 👋 

Here is the WIP slidedeck for the engineering onboarding meeting:

https://docs.google.com/presentation/d/16GgE7__d8XvjIVTfipCcL9IpJz4Ko1rw1B3ETMgrVds/edit?usp=sharing

I set the link to be freely editable.  See y'all on June 25th!

**Comment by gunsch** [comment]
Looks great! I think it's a little long (trying to get through a 50-slide deck in under an hour scares me), but probably if we just combine a few of those slides and speak to the high-level guidance we'll be fine.

I'll edit it a bit if we need it before you're back, otherwise this looks great for now. Thanks!

**Comment by hpjaj** [comment]
@rtluu @ricetj - I spoke with @wyattwalter today, and we'll want to make sure that the Operations team and Support team are on the same page with what is being delivered to new VFS teams during the onboarding process.   Want to ensure that there are not duplicative efforts, consistent documentation delivered to VFS teams, and a common message being delivered during the onboard process.

One piece of this is what is presented during the engineering onboarding meeting.  There is a link to the slide deck above.  Some questions would be:

- Can/should a representative from the operations team be part of the engineering onboarding meeting (or perhaps one of the follow up meetings during the week)?
- Is there any Operations related slides that should be added to this deck?  If yes, who can we work with on that?

**Comment by hpjaj** [comment]
After confirming with @accbjt and @1Copenut , here is who will be presenting what:

- Welcome - Trevor
- Introductions - Trevor
- Getting Setup - Bill
- VSP Engineering Best Practices - Bill
- Ticktes/Issues/User Stories - Bill
- Pull Requests - Bill (or new BE dev)
- Process - Bill (or new BE dev)
- Backend Development - Bill (or new BE dev)
- Frontend Development - Bill
- 508/Accessibility - Trevor
- General - Bill
- Resources - Bill

**Comment by hpjaj** [comment]
@accbjt @1Copenut - Here is the agenda for the meeting, as it stands today.  I am dropping it in here as a comment as I am not sure how you all will be sharing this agenda with the VFS teams. Once all of the agendas are done, [this ticket](https://github.com/department-of-veterans-affairs/va.gov-team/issues/417) is there to format them in a common manner.

Please use this as a basis to get started, and update as you see fit!

## Meeting Goals

The goals of the engineering on-boarding meeting are:

- For the Platform product support engineers and the on-boarding engineers to meet one another.
- To introduce the on-boarding engineers to the Platform engineering best practices and guidelines.
- To provide the on-boarding engineers with resources that they will need during on-boarding and beyond.
- To provide direction and answer any initial questions.

## Agenda Topics

- Introductions
- Getting setup
- VSP engineering best practices
- Resources
- Feedback
- Q & A

**Comment by hpjaj** [comment]
Hey @accbjt @1Copenut -  Here are all my notes:

- There is probably too much content being shared in the best practices section
  - Go back through and decide which ones are most important and relevant on day 1 (and planting seeds), and trim out the rest.  
  - Letting them know that all of our best practices are in that markdown doc 
- In the all team meeting, should be a layout of what the first, second, third, etc. week's schedule and itinerary looks like
- Should reiterate the engineering related activities at the beginning, so they know that this talk will be "high level", and the follow up activities and work will be more detailed for each topic
- Is this deck going to be shared with the VFS team?  i.e. PDF, link
  - If yes, let them know that at the beginning so they are more comfortable listening, and not frantically taking notes, and perhaps missing some things that are said
- Setup
  - Talk more here
- At the beginning of the best practices session, show the best practices mark down doc, and say that this is the doc that we are going to review
  - Do this immediately before the "motivation" slide; so first thing
  - After this, we will be working with you on all of this, we don't expect you to remember all of this, just setting the stage about things we'll be highlighting as you 
- Visuals: 
  - Have a sample issues/ticket to speak to 
  - Example of an API contract in an issue 
  - Image of the @mentions for the FE and BE groups
  - Of the ticket for the status details slide
  - In the "deployed every weekday" could show an image from slack that we get everyday at 2pm ET
  - Have a sample PR to speak to for the PR section
  - For a lot of these images, when possible, includes links to actual items in the "resources" section (i.e. link to the issue in question, the PR in question, etc.)
  - Screenshot of swagger docs in a PR 
  - By the by, you can come through my closed PR's and issues for these examples
  - Show image of a WIP PR in the title of a PR
- Change link on PR re-request to go directly to the re-request in GitHub's help docs
- Remove the slides in the process section about "connect with devops if you are integrating an external API"
- PR slide
  - bold the "no less than once a week" in that slide around PR frequency
- 508 section
  - Try to speak at a higher level, and let them know that we will have opportunities soon to dig into these details, specific to their code, and that you just wanted to highlight that these are some of the areas that are important to us

Good luck!!

**Comment by accbjt** [comment]
Here is what I took from the dry run meeting
- Change best practice to patterns
- Don't try teach them how to program but let them know this is what has worked for us
- Try not to get into too much detail and make it more higher level
- Add agenda
- Reiterating that we are a team was great, should mention this in the beginning

**Comment by 1Copenut** [comment]
Just a few more items not already covered above:
* 508, higher level. I'm a tactical details thinker, so this will be an iterative task.
* Make cross-references from in-slide images to resources and vice-versa where they'll add impact.
* Mention we're making a PDF as a takeaway.

**Comment by andreahewitt-odd** [comment]
@smoonpeyes Can you move the deck to github?

**Comment by smoonpeyes** [comment]
Presentation added: https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/Platform/Onboarding/templates/VFS-Engineering-Onboarding.pptx
